### PR TITLE
Fix various issues with the release staging script

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

**Comment by t8m** [comment]
ping @openssl/otc

File: release-tools/release-aux/release-data-fn.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@
 
 # Public or premium release?  Let the version numbers determine it!
 declare -A _ossl_release_types=(
-    [premium]='^1\.0\.2'
-    [public]='^(1\.1\.1|[3-9]\.)'
+    [premium]='^1\.[01]\.[12]'
```
Comment (nhorman) [inline release-tools/release-aux/release-data-fn.sh:11]: This to me (I think) says premium releases are identified as 1.0.1, 1.0.2, 1.1.1 and 1.1.2

Should that be future proofed for a bit by extending [12] to [123456789] or some such?  Presumably, we're going to bump these patch levels, or are we using alphabetic designation in perpetuity here?

**Review by nhorman — COMMENTED** [review]
Generally looks good, just the one question

File: release-tools/release-aux/release-data-fn.sh (hunk 1 above)
Comment (t8m) [reply release-tools/release-aux/release-data-fn.sh:11]: We IMO should not bump the number as that could possibly break things for our customers depending on the fix being marked by the alphabetic suffix. There will be a problem when we reach 1.0.2zzzzzzzzzze release :grin: as that will be the last one that can be represented by the OPENSSL_VERSION_NUMBER. However that will be reached some time near the end of this century if we keep the current rate of patch releases.

**Review by nhorman — APPROVED** [review]

File: release-tools/release-aux/release-data-fn.sh (hunk 1 above)
Comment (levitte) [inline release-tools/release-aux/release-data-fn.sh:11]: Why not just `'1\.'`?

Comment (levitte) [reply release-tools/release-aux/release-data-fn.sh:11]: I really don't see why we should make it more complicated than necessary...

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
Merged. Thank you for the reviews.

**Pull request closed without merging**